source/deprecations/v1.x.html.md

```diff
@@ -1339,3 +1373,48 @@ export default Ember.Helper.helper(function([firstArg, secondArg], hash) {
   // helper code
 });
 ```
+
+#### Using `TrackedArray` or `SubArray`
+
+`TrackedArray` and `SubArray` were internal classes used as implementation
+details of `ArrayComputed`.  `ArrayComputed` is deprecated to be removed
+in version 2.0.0.  As a result, these 2 private classes will be removed as
+well.
+
+#### Using `@each` as a leaf node in a dependent key
+Using `@each` at the end of a computed key is deprecated and will not work in
+Ember 2.0
+
+```javascript
+invalid: Ember.computed('myObject.@each', function () {
+  //no longer valid for monitoring changes to arrays
+});
+```
+
+When defining dependent keys for computed properties, ember 2.0+ will treat
+`@each` and `[]` differently.
+
+`@each` will monitor specific properties within an array of objects.
+
+```javascript
+eachProp: Ember.computed('myObj.posts.@each.title', function () {
+  //fired whenever one of the blog post titles is changed.
+});
+```
+
+`[]` will monitor mutations to the array.
+
+```javascript
+arrProp: Ember.computed('myObj.posts.[]', function () {
+  //fired whenever a blog post is added or removed
+});
+```
+
+#### Overriding `render` When Extending a Component or View.
+
+The `render` method on components and views should not be overridden and will
+go away in Ember 2.x.  Modifications to Ember rendering should be made by
+overriding Ember's new
+[component lifecycle hooks](http://emberjs.com/blog/2015/06/12/ember-1-13-0-released.html#toc_component-lifecycle-hooks)
```

Comment:
Pointing users overriding render to the new lifecycle hooks, but would like validation on whether this is an appropriate alternative to point them at.